src/log.rs

```diff
@@ -162,4 +165,76 @@ impl Writer {
 
         Ok(())
     }
+
+    pub fn compact_log<S: Serialize, P: AsRef<Path>>(
+        &self,
+        path: P,
+        data: Vec<S>,
+    ) -> Result<(), Error> {
+        let new_db_path = path.as_ref().with_file_name(Uuid::new_v4().to_string());
```

Comment:
I don't think we need `uuid` for this, you can just append `.temp` to this. You can overwrite any prior log compacting events.

If you wanted to make a case for not overwriting you could use the current timestamp as a file unique-ifier.